.github/workflows/ci.yml

```diff
@@ -131,14 +131,194 @@ jobs:
+        cat verify-${{ matrix.framework }}.config | briefcase new
+    - name: Build Test App
+      run: |
+        cd tests/apps/verify-${{ matrix.framework }}
+        briefcase create
+        briefcase build
+        briefcase package --adhoc-sign
+    - name: Build Test App using Xcode Template
+      run: |
+        cd tests/apps/verify-${{ matrix.framework }}
+        briefcase create macOS Xcode
+        briefcase build macOS Xcode
+        briefcase package macOS Xcode --adhoc-sign
+
+  verify-desktop-windows:
+    name: Windows desktop app verification
+    needs: platform-compat
+    strategy:
+      max-parallel: 4
+      matrix:
+        framework: ['toga', 'pyside2', 'pyside6', 'ppb']
+    runs-on: 'windows-latest'
+    steps:
+    - name: Cache Briefcase tools
+      uses: actions/cache@v3
+      with:
+        key: briefcase-windows-latest
+        path: |
+          ~/.cookiecutters
+          ~\AppData\Local\BeeWare\briefcase\Cache
+          ~\AppData\Local\pip\Cache
+          C:\ProgramData\DockerDesktop
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v3.1.2
+      with:
+        python-version: ${{ env.python_version }}
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip
+        pip install --upgrade setuptools
+        pip install .
+    - name: Create Test app
+      run: |
+        cd tests/apps
+        cat verify-${{ matrix.framework }}.config | briefcase new
+    - name: Build Test App
+      run: |
+        cd tests/apps/verify-${{ matrix.framework }}
+        briefcase create
+        briefcase build
+        briefcase package
+    - name: Build Test App with Visual Studio Template
+      run: |
+        cd tests/apps/verify-${{ matrix.framework }}
+        briefcase create windows VisualStudio
+        briefcase build windows VisualStudio
+        briefcase package windows VisualStudio
+
+  verify-desktop-linux:
+    name: Linux desktop app verification
+    needs: platform-compat
+    strategy:
+      max-parallel: 4
+      matrix:
+        framework: ['toga', 'pyside2', 'pyside6', 'ppb']
+    # Need to use at least 22.04 to get the bugfix in flatpak that doesn't handle spaces in filenames.
+    runs-on: 'ubuntu-22.04'
+    steps:
+    - name: Cache Briefcase tools
+      uses: actions/cache@v3
+      with:
+        key: briefcase-ubuntu-latest
+        # cache action cannot cache docker images (actions/cache#31) - /var/lib/docker
+        path: |
+          ~/.cookiecutters
+          ~/.cache/briefcase
+          ~/.cache/pip
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v3.1.2
+      with:
+        python-version: ${{ env.python_version }}
+    - name: Install system dependencies
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y python3-gi python3-gi-cairo gir1.2-gtk-3.0 python3-dev libgirepository1.0-dev libcairo2-dev pkg-config flatpak flatpak-builder
```

Comment:
I'm fairly sure we do - these packages aren't in the [list of apt packages in the 22.04 runner config](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#installed-apt-packages). However, I'll experiment and see.

Reply:
Oh - wait - now I see what you're saying: we don't need these because we're building *in* docker. You're completely correct - we only need the flatpak tools.